src/main/java/graphql/execution2/result/ExecutionResultMultiZipper.java

```diff
@@ -68,7 +73,7 @@ public ExecutionResultMultiZipper withReplacedZipper(ExecutionResultZipper oldZi
 
     private List<ExecutionResultZipper> getDeepestZippers(List<ExecutionResultZipper> zippers) {
         Map<Integer, List<ExecutionResultZipper>> grouped =
-                zippers.stream().collect(Collectors.groupingBy(executionResultZipper -> executionResultZipper.getBreadcrumbList().size()));
+                zippers.stream().collect(Collectors.groupingBy(executionResultZipper -> executionResultZipper.getBreadcrumbList().size(), LinkedHashMap::new, mapping(Function.identity(), toList())));
         Integer maxLevel = Collections.max(grouped.keySet());
```

Comment:
Almost deserves it is on `Collector` based puirely on how much boilerplate.  Perhaps its own method

`zippers.stream().collect(Collectors.groupingBy(thisGrouping(executionResultZipper -> executionResultZipper.getBreadcrumbList().size()));`